include/envoy/http/filter.h

```diff
@@ -33,7 +33,15 @@ enum class FilterHeadersStatus {
   StopIteration,
   // Continue iteration to remaining filters, but ignore any subsequent data or trailers. This
   // results in creating a header only request/response.
-  ContinueAndEndStream
+  ContinueAndEndStream,
+  // Stop iterations for headers as well as data and trailers that follows the headers. Returning
+  // StopAllTypesIteration means the filter that returns the status and all the filters following
+  // that filter stop processing. The filters before are not impacted. continueDecoding() MUST be
+  // called if continued filter iteration is desired. To avoid buffering large amount of data, a
+  // rate-limiting filter can be added before the filter that can return StopAllTypesIteration.
+  // Only used in decoding path.
```

Comment:
would be nice to add a line on the use case for this type of status

Reply:
Fixed. Thanks for the feedback.